http/src/config.rs

```diff
@@ -146,10 +130,28 @@ fn create(
         },
     };
 
-    serde_json::to_writer_pretty(BufWriter::new(config), &config_contents)
+    let config_path = ipfs_path.join("config");
+
+    let config_file = fs::create_dir_all(&ipfs_path)
+        .map_err(InitializationError::DirectoryCreationFailed)
+        .and_then(|_| {
+            fs::File::create(&config_path).map_err(InitializationError::ConfigCreationFailed)
```

Comment:
Probably an older bug: Will [`std::fs::File::create`](https://doc.rust-lang.org/stable/std/fs/struct.File.html#method.create) truncate the target? Yes. We should use [`OpenOptions::create_new`](https://doc.rust-lang.org/stable/std/fs/struct.OpenOptions.html#method.create_new) to avoid truncation. The only way the file exists at this point if it was created between src/main.rs checking `!file.exists()` and this create, but we still shouldn't truncate it. The `!file.exists()` from src/main.rs can probably stay, it's a nice sanity check before rsa keygen.